pyintegration/deephaven2/table_listener.py

```diff
@@ -0,0 +1,270 @@
+           start_listening: bool = True, replay_initial: bool = False,
+           lock_type="shared") -> TableListenerHandle:
+    """ Listen to table changes.
+
+    Table change events are processed by listener, which can be either (1) a callable (e.g. function) or
+    (2) an object which provides an "onUpdate" method.
+    In either case, the method must have one of the following signatures.
+
+    * (added, removed, modified): shift_oblivious or legacy
+    * (isReplay, added, removed, modified): shift_oblivious + replay
+    * (update): shift-aware
+    * (isReplay, update): shift-aware + replay
+
+    For legacy listeners, added, removed, and modified are the indices of the rows which changed.
+    For shift-aware listeners, update is an object which describes the table update.
+    Listeners which support replaying the initial table snapshot have an additional parameter, inReplay, which is
+    true when replaying the initial snapshot and false during normal updates.
+
+    See the Deephaven listener documentation for details on processing update events.  This documentation covers the
+    details of the added, removed, and modified row sets; row shift information; as well as the details of how to apply
+    the update object.  It also has examples on how to access current and previous-tick table values.
+
+    Args:
+        t (Table): table to listen to
+        listener (Callable): listener to process changes
+        description (str): description for the UpdatePerformanceTracker to append to the listener's entry description
+        retain (bool): whether a hard reference to this listener should be maintained to prevent it from being
+            collected, default is True
+        listener_type (str): listener type, valid values are "auto", "legacy", and "shift_aware"
+            "auto" (default) uses inspection to automatically determine the type of input listener
+            "legacy" is for a legacy listener, which takes three (added, removed, modified) or four (isReplay, added,
+                removed, modified) arguments
+            "shift_aware" is for a shift-aware listener, which takes one (update) or two (isReplay, update) arguments
+        start_listening (bool): True to create the listener and register the listener with the table. The listener will
+            see updates. False to create the listener, but do not register the listener with the table. The listener
+            will not see updates. default is True
+        replay_initial (bool): True to replay the initial table contents to the listener. False to only listen to new
+            table changes. To replay the initial image, the listener must support replay. default is False
+        lock_type (str): UGP lock type, used when replay_initial=True, valid values are 'exclusive' and 'shared'.
+
+    Returns:
+        a TableListenerHandle instance
+
+    Raises:
```

Comment:
```suggestion
    """ Listen to table changes.

    Table change events are processed by `listener`, which can be either (1) a callable (e.g. function) or
    (2) an object that provides an "onUpdate" method.
    In either case, the method must have one of the following signatures.

    * (added, removed, modified): shift_oblivious / legacy
    * (isReplay, added, removed, modified): shift_oblivious / legacy + replay
    * (update): shift-aware
    * (isReplay, update): shift-aware + replay

    For legacy shift-oblivious listeners, `added`, `removed`, and `modified` are the indices of the rows that changed.
    For shift-aware listeners, `update` is an object that describes the table update.
    Listeners that support replaying the initial table snapshot have an additional parameter, `inReplay`, which is
    true when replaying the initial snapshot and false during normal updates.

    See the Deephaven listener documentation for details on processing update events.  This documentation covers the
    details of the added, removed, and modified row sets; row shift information; as well as the details of how to apply
    the update object.  It also has examples on how to access current and previous-tick table values.

    Args:
        t (Table): table to listen to
        listener (Callable): listener for table changes
        description (str): description for the UpdatePerformanceTracker to append to the listener's entry description
        retain (bool): whether a hard reference to this listener should be maintained to prevent it from being
            collected, default is True
        listener_type (str): listener type, valid values are "auto", "legacy", and "shift_aware"
            "auto" (default) uses inspection to automatically determine the type of input listener
            "legacy" is for a legacy listener, which takes three (added, removed, modified) or four (isReplay, added,
                removed, modified) arguments
            "shift_aware" is for a shift-aware listener, which takes one (update) or two (isReplay, update) arguments
        start_listening (bool): True to create the listener and register the listener with the table. The listener will
            see updates. False to create the listener, but do not register the listener with the table. The listener
            will not see updates. Default is True
        replay_initial (bool): True to replay the initial table contents to the listener. False to only listen to new
            table changes. To replay the initial image, the listener must support replay. Default is False
        lock_type (str): UGP lock type, used when replay_initial=True, valid values are "exclusive" and "shared".  Default is "shared".

    Returns:
        a TableListenerHandle instance

    Raises:
```

Should we support `listener_type` of "shift_oblivious" and then note that "legacy" is an alias to it?

Reply:
this is v2, I think we should just drop 'legacy'.